--- ../src-base/minecraft/net/minecraft/world/chunk/storage/ExtendedBlockStorage.java
+++ ../src-work/minecraft/net/minecraft/world/chunk/storage/ExtendedBlockStorage.java
@@ -6,6 +6,8 @@
 import net.minecraft.world.chunk.BlockStateContainer;
 import net.minecraft.world.chunk.NibbleArray;
 
+import carpet.CarpetSettings;
+
 public class ExtendedBlockStorage
 {
     private final int field_76684_a;
@@ -63,7 +65,12 @@
 
     public boolean func_76663_a()
     {
-        return this.field_76682_b == 0;
+    	// NewLight PHIPRO-CARPET
+    	if (CarpetSettings.newLight){
+    		return false;
+    	}else{
+    		return this.field_76682_b == 0;
+    	}
     }
 
     public boolean func_76675_b()
@@ -78,6 +85,7 @@
 
     public void func_76657_c(int p_76657_1_, int p_76657_2_, int p_76657_3_, int p_76657_4_)
     {
+        if(CarpetSettings.noLightUpdates) return;
         this.field_76685_h.func_76581_a(p_76657_1_, p_76657_2_, p_76657_3_, p_76657_4_);
     }
 
@@ -88,6 +96,7 @@
 
     public void func_76677_d(int p_76677_1_, int p_76677_2_, int p_76677_3_, int p_76677_4_)
     {
+        if(CarpetSettings.noLightUpdates) return;
         this.field_76679_g.func_76581_a(p_76677_1_, p_76677_2_, p_76677_3_, p_76677_4_);
     }
 
@@ -147,4 +156,8 @@
     {
         this.field_76685_h = p_76666_1_;
     }
+
+    public BlockStateContainer getBlockStateContainer() {
+        return field_177488_d;
+    }
 }
